### PR TITLE
Offsets

### PR DESCRIPTION
**Pull request by ThePrimeagen**
THE OFFSETS

### CODE REVIEW

**Comment by ThePrimeagen** [comment]
@brandoncc @asbjornhaland if there is anything I have missed. lmk.

So easy to miss things :)

**Comment by brandoncc** [comment]
Nothing jumps out at me. I'll try it out and see if I can find any bugs. Nice work!

File: plugin/mark.vim [hunk 1]
```diff
@@ -1,5 +1,10 @@
+fun! HarpoonLeave()
```
Comment (ThePrimeagen) [inline plugin/mark.vim:1]: At some point, we could probably kill this file and the other .vim file.

Just use vim.cmd [[ ... ]]

File: lua/harpoon/mark.lua [hunk 2]
```diff
@@ -103,41 +112,51 @@ M.add_file = function(file_name_or_buf_id)
...
-    end
+    local ok, res = pcall(function()
+        local buf_name = get_buf_name()
+        local idx = M.get_index_of(buf_name)
+        if not M.valid_index(idx) then
+            return
+        end
 
-    local line = vim.api.nvim_eval("line('.')");
+        harpoon.get_mark_config().marks[idx].row =
```
Comment (asbjornhaland) [inline lua/harpoon/mark.lua:152]: Are `col` ever set? If it's not needed, I think it should be removed from `create_mark()`?

**Review by asbjornhaland — COMMENTED** [review]
I'm not available to test this right now, but could not see anything that stood out (except my one comment) when I skimmed through the code. Good work!

File: plugin/mark.vim (hunk 1 above)
Comment (asbjornhaland) [reply plugin/mark.vim:1]: I agree.

File: lua/harpoon/mark.lua (hunk 2 above)
Comment (ThePrimeagen) [reply lua/harpoon/mark.lua:152]: I want to get col to work, I just haven't yet.  I am going to keep it in there until I figure that part out :)

Comment (asbjornhaland) [reply lua/harpoon/mark.lua:152]: `:lua print(vim.fn.col("."))`

**Pull request merged**